store/mongo/datastore_mongo.go

```diff
@@ -603,6 +603,49 @@ func (db *DataStoreMongo) AddAuthSet(ctx context.Context, set model.AuthSet) err
 	return nil
 }
 
+// UpsertAuthSetStatus inserts a new auth set and if it exists, ensures the
+// AuthSet.Status matches the one in authSet.
+func (db *DataStoreMongo) UpsertAuthSetStatus(ctx context.Context, authSet *model.AuthSet) error {
```

Comment:
general comment: AddAuthSet does a lot of the same, this means that now we have duplicate code for some of that, which can lead to problems in maintaining the code and can be cause for regressions. I would try to create a single place to upsert an authset.

Reply:
I put the shared code into a new function for populating the missing fields in the AuthSet.